Sent Component and Send Chart Component designs are ready

NataliaMilgevskaya opened the pull request:
Sent Component and Send Chart Component designs are ready

desousamario85 committed:
Add or update the Azure App Service build and deployment workflow config

desousamario85 committed:
Update master_zappie-dev.yml

BenGWeeks committed:
Merge conflicts

BenGWeeks committed:
Converted to TypeScript react web app from JavaScript

BenGWeeks committed:
Static FeedComponent added to a homepage

BenGWeeks committed:
Code refactoring and updates

- Renamed the "components" folder to "services"
- Moved a lot of code in teamsBot.ts to the Commands folder
- Updated a lot of the LNBits service methods to be clear if using inKey or adminKey (still some work needed here).
- Updated the tabs so we now have a feed of zaps (still needs work - we need a way to identify zaps - probably add additional metadata to the invoice either in memo or using the "extra" property - can see how to read, not amend).

BenGWeeks committed:
Updated tabs

BenGWeeks committed:
Updated Feed table

- From/To now added to the createInvoice.
- From/To now displayed on the FeedList.
- Zap.from and Zap.to now set to just strings (maybe in the future this should be a User object).
- Create a sinple Utility function for wallets.

BenGWeeks committed:
Reduced console logs

We don't need so many. Should use debugger more than console logging anyway.

BenGWeeks committed:
Resolved the date period selection

- User can now select 7-days, 30-days or 60-days and timestamp is amended correctly (was setting a timestamp in the future).
- The styling of 7-days, 30-days or 60-days is now styled accordingly.
- Defaults to 7-days

BenGWeeks committed:
Resolved the date period selection

- User can now select 7-days, 30-days or 60-days and timestamp is amended correctly (was setting a timestamp in the future).
- The styling of 7-days, 30-days or 60-days is now styled accordingly.
- Defaults to 7-days

BenGWeeks committed:
Merge branch 'react-web-app' of https://github.com/BenGWeeks/Zapp.ie into react-web-app

BenGWeeks committed:
Zap feed ordering

- Add a zap icon to the feed.
- Added a datetime column to the zaps
- Ordered the zaps with newest at the top

desousamario85 commented on tabs/src/Home.tsx:
```diff
@@ -1,11 +1,21 @@
 
 const Home: React.FC = () => {
   return (
-    <div>
```
Please add some text on the page.

desousamario85 reviewed (changes requested):
Code is not working, please review.

BenGWeeks committed:
Merge branch 'main' into react-web-app

BenGWeeks committed:
Delete App.test.tsx

BenGWeeks committed:
React web app (#9)

* Added React webapp

* Fix trailing whitespace errors in manifest.json

* Fix trailing whitespace errors in manifest.json

* Updated Readme

* Add or update the Azure App Service build and deployment workflow config

* Update master_zappie-dev.yml

* Converted to TypeScript react web app from JavaScript

* Static FeedComponent added to a homepage

* Code refactoring and updates

- Renamed the "components" folder to "services"
- Moved a lot of code in teamsBot.ts to the Commands folder
- Updated a lot of the LNBits service methods to be clear if using inKey or adminKey (still some work needed here).
- Updated the tabs so we now have a feed of zaps (still needs work - we need a way to identify zaps - probably add additional metadata to the invoice either in memo or using the "extra" property - can see how to read, not amend).

* Updated tabs

* Updated Feed table

- From/To now added to the createInvoice.
- From/To now displayed on the FeedList.
- Zap.from and Zap.to now set to just strings (maybe in the future this should be a User object).
- Create a sinple Utility function for wallets.

* Reduced console logs

We don't need so many. Should use debugger more than console logging anyway.

* Resolved the date period selection

- User can now select 7-days, 30-days or 60-days and timestamp is amended correctly (was setting a timestamp in the future).
- The styling of 7-days, 30-days or 60-days is now styled accordingly.
- Defaults to 7-days

* Resolved the date period selection

- User can now select 7-days, 30-days or 60-days and timestamp is amended correctly (was setting a timestamp in the future).
- The styling of 7-days, 30-days or 60-days is now styled accordingly.
- Defaults to 7-days

* Zap feed ordering

- Add a zap icon to the feed.
- Added a datetime column to the zaps
- Ordered the zaps with newest at the top

* Delete App.test.tsx

---------

Co-authored-by: Mario DeSousa <Mario.DeSousa@evros.ie>
Co-authored-by: desousamario85 <53436812+desousamario85@users.noreply.github.com>

BenGWeeks committed:
Delete lnbitService.test.ts

Will look at this later.

BenGWeeks committed:
Merge remote-tracking branch 'origin/main' into react-web-app

BenGWeeks committed:
Minor fix

BenGWeeks committed:
Added a FooterComponent

BenGWeeks committed:
Initial commit

Added the zap activity chart.
Updated types.
Updated methods.
Updated middleware.

BenGWeeks committed:
Ignore Deleted Wallets, sendZap fixes & MiddleWare correction

We now ignore deleted wallets (that are still in the backend).
SendZaps now sends correctly.
Middleware doesn't keep firing off (still too much though but was causing SendZap to loop for some reason).

BenGWeeks committed:
Proxy updates

BenGWeeks committed:
Resolving auth token issue

Looks like repeated calls get blocked by the browser.

akash2017sky committed:
Wallet allowance component (#15)

* Allowance component files added

* UPdated the code to fetch the data from Service.

* code formatting and bug fixing.

* Send and recieve color

BenGWeeks committed:
Removed lnbitsServiceLocal.ts pointer

BenGWeeks committed:
Resolved failed to fetch issue

BenGWeeks committed:
Deleted lnbitsService.ts and modified lnbitsServiceLocal.ts

NataliaMilgevskaya committed:
Rewards Component. Gets data from lnbits Marketplace (#11)

* Rewards Component. Gets data from lnbits Marketplace

* Minor updates

- Fixed .gitignore
- Returned Home to before.
- Created a new Rewards page.

* Resolving auth token issue

Looks like repeated calls get blocked by the browser.

* User Story 90317: As a user, I want to see the available rewards

* WIP: Changes to RewardsComponent.tsx

---------

Co-authored-by: Ben Weeks <ben.weeks@outlook.com>

BenGWeeks committed:
Sorted out types

BenGWeeks committed:
Removed lnbitsService.ts pointer

BenGWeeks committed:
Merged latest from origin/main

BenGWeeks committed:
Removed lnbitsService.ts

BenGWeeks committed:
Tweaked Activity chart period to fit
